File: java/P102.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> answer = new ArrayList<List<Integer>>();
        if (root == null) return answer;
        Queue<TreeNode> queue = new LinkedList<TreeNode>();
        queue.offer(root);
        while (!queue.isEmpty()) {
            int levelsize = queue.size();
            List<Integer> tempqueue = new ArrayList<Integer>();
            for (int i = 0; i < levelsize; i++) {
                if (queue.peek().left != null) queue.offer(queue.peek().left);
                if (queue.peek().right != null) queue.offer(queue.peek().right);
                tempqueue.add(queue.poll().val);
            }
            answer.add(tempqueue);
        }
        return answer;
    }
}

File: python/P102.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrder(self, root):
        if not root:
            return []
        this_level = [root]
        this_level_val = []
        next_level = []
        answer = []
        while this_level:
            while this_level:
                current = this_level.pop(0)
                this_level_val.append(current.val)
                if current.left:
                    next_level.append(current.left)
                if current.right:
                    next_level.append(current.right)
            answer.append(this_level_val)
            this_level_val = []
            this_level = next_level
            next_level = []
        return answer
        